dex/src/matching.rs

```diff
@@ -368,23 +382,45 @@ impl<'ob> OrderBookState<'ob> {
                 .as_leaf_mut()
                 .unwrap();
 
+            let order_id = best_bid_ref.order_id();
+
+            if best_bid_ref.is_order_expired(epoch_start_ts, start_epoch_seq_num, current_ts, true)
+            {
+                let order = self.orders_mut(Side::Bid).remove_by_key(order_id).unwrap();
+                let out = Event::new(EventView::Out {
+                    side: Side::Bid,
+                    release_funds: true,
+                    native_qty_unlocked: order.quantity() * order.price().get() * pc_lot_size,
+                    native_qty_still_locked: 0,
+                    order_id: order.order_id(),
+                    owner: order.owner(),
+                    owner_slot: order.owner_slot(),
+                    client_order_id: NonZeroU64::new(order.client_order_id()),
+                });
+                event_q
+                    .push_back(out)
+                    .map_err(|_| DexErrorCode::EventQueueFull)?;
```

Comment:
should this be a different error? same below for the new_bid function

Reply:
this error is correct its just the error that happens when the event_q is full when trying to add an event to it

Reply:
ahhh the fancy syntax got me, youre right